### PR TITLE
Fix: Only if `.sav` file has changes a recovery dialog is shown

### PR DESCRIPTION
**Pull request by koppor**
When JabRef is forced-closed and restarted, it should not nag about a `.sav` file.

The issue described at https://github.com/jabref/jabref/issues/6109 seems to be a larger one. We will that investigate later.

### CODE REVIEW

File: src/main/java/org/jabref/logic/autosaveandbackup/BackupManager.java [hunk 1]
```diff
@@ -86,13 +87,27 @@ public static void shutdown(BibDatabaseContext bibDatabaseContext) {
...
-    public static boolean checkForBackupFile(Path originalPath) {
+    public static boolean backupFileDiffers(Path originalPath) {
         Path backupPath = getBackupPath(originalPath);
-        return Files.exists(backupPath) && !Files.isDirectory(backupPath);
+        if (!Files.exists(backupPath) || Files.isDirectory(backupPath)) {
+            return false;
+        }
+
+        try {
+            return !com.google.common.io.Files.equal(originalPath.toFile(), backupPath.toFile());
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/logic/autosaveandbackup/BackupManager.java]: Files.mismatch should do the job
https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/nio/file/Files.html#mismatch(java.nio.file.Path,java.nio.file.Path)

File: src/test/java/org/jabref/logic/autosaveandbackup/BackupManagerTest.java [hunk 2]
```diff
@@ -15,4 +17,22 @@ public void backupFileNameIsCorrectlyGeneratedWithinTmpDirectory() {
...
+
+    @Test
+    public void backupFileIsEqual() throws Exception {
+        Path originalFile = Path.of(BackupManagerTest.class.getResource("no-changes.bib").toURI());
+        assertFalse(BackupManager.backupFileDiffers(originalFile));
+    }
+
+    @Test
+    public void backupFileDiffers() throws Exception {
+        Path originalFile = Path.of(BackupManagerTest.class.getResource("changes.bib").toURI());
```
Comment (tobiasdiez) [inline src/test/java/org/jabref/logic/autosaveandbackup/BackupManagerTest.java]: How do these tests work? I cannot see a `.sav` file in the folder?!

Comment (stefan-kolb) [reply src/test/java/org/jabref/logic/autosaveandbackup/BackupManagerTest.java]: Save files are missing in the commit! @tobiasdiez Why do you approve the PR if you have objections (that are correct!)?

Comment (koppor) [reply src/test/java/org/jabref/logic/autosaveandbackup/BackupManagerTest.java]: They did not work - I added the missing files in ee472cf

grafik

**Pull request merged**